client/packages/invoices/src/InboundShipment/DetailView/modals/InboundLineEdit/TabTables.tsx

```diff
@@ -103,27 +122,94 @@ export const QuantityTableComponent: FC<TableProps> = ({
       setter: updateDraftLine,
     });
   }
-  columnDefinitions.push(
-    [
-      'numberOfPacks',
-      {
-        Cell: NumberOfPacksCell,
-        width: 100,
+
+  if (displayInDoses) {
+    columnDefinitions.push({
+      key: 'dosesPerPack',
+      label: `${t('label.doses-per')} ${itemUnitName}`,
+      width: 120,
+      align: ColumnAlign.Right,
+      accessor: ({ rowData }) => rowData.item?.doses,
+    });
+  }
+
+  if (shouldShowNumberOfPacks) {
```

Comment:
Starting to get quite hard to tell what columns should be showing from this file, lots of different if then push - I wonder if a nice refactor could be to put all column definitions in functions (like batch and expiry) and then in here, we define just `if (case) return [these columns]` ... 🤷‍♀️

Reply:
Yup, will write up an issue